### PR TITLE
TF-M 2.0.0 + Mbed TLS 3.5.2 upmerge+ Oberon PSA core 1.2.1 + ocrypto 3.0.14 + cc3xx library 0.9.19 update

### PR DESCRIPTION
**Pull request by mswarowsky**
test-sdk-nrf: test-nrf-pr-13662

### CODE REVIEW

File: applications/serial_lte_modem/overlay-native_tls.conf [hunk 1]
```diff
@@ -57,9 +59,11 @@ CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED=y
 CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED=y
 # Enable RSA
 CONFIG_MBEDTLS_KEY_EXCHANGE_RSA_ENABLED=y
-CONFIG_MBEDTLS_RSA_C=y
```
Comment (MarkusLassila) [inline applications/serial_lte_modem/overlay-native_tls.conf:60]: @mswarowsky: Apologies, this is still needed. Blaming this one to lack of sleep. :)

Comment (mswarowsky) [reply applications/serial_lte_modem/overlay-native_tls.conf:60]: No worries, thanks for correcting

**Review by SeppoTakalo — APPROVED** [review]

File: subsys/nrf_security/src/legacy/CMakeLists.txt [hunk 2]
```diff
@@ -165,3 +177,23 @@ target_include_directories(${mbedcrypto_target}
 INTERFACE
   $<$<TARGET_EXISTS:nrf_cc3xx_platform>:$<TARGET_PROPERTY:nrf_cc3xx_platform,INTERFACE_INCLUDE_DIRECTORIES>>
 )
+
+if(NOT is_getting_includes)
+  if(NOT DEFINED ARM_MBEDTLS_PATH)
+	message(FATAL_ERROR "misconfiguration")
```
Comment (frkv) [inline subsys/nrf_security/src/legacy/CMakeLists.txt:183]: Should be 2 space indentation, not tabs

Same on target_include_directories further down

**Review by plskeggs — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Review by maciejbaczmanski — COMMENTED** [review]
Wider test scope for OpenThread: https://jenkins-ncs.nordicsemi.no/job/latest/job/night/job/test-fw-nrfconnect-thread/job/thread-test-branch/426/
I'll let you know when tests have finished

**Review by Vge0rge — APPROVED** [review]

**Review by mia-ko — APPROVED** [review]

**Review by vili-nordic — APPROVED** [review]

**Review by magnev — APPROVED** [review]

**Pull request merged**